modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -381,40 +426,65 @@
                                        {$size "$count"} ; HACK
                                        true)])))
 
+(defmulti ^:private expand-aggregation (fn [[ag-type arg]]
+                                         (if (= ag-type :named)
+                                           (recur arg)
+                                           ag-type)))
+
+(defmethod expand-aggregation :share
+  [[_ pred :as ag]]
+  (let [count-if-name (name (gensym "count-if"))
+        count-name    (name (gensym "count-"))
+        pred          (if (= (first pred) :share)
+                        (second pred)
+                        pred)]
+    [[[count-if-name (aggregation->rvalue [:count-if pred])]
+      [count-name (aggregation->rvalue [:count])]]
+     [[(annotate/aggregation-name ag) {$divide [(str "$" count-if-name) (str "$" count-name)]}]]]))
+
+(defmethod expand-aggregation :default
+  [ag]
+  (println ag)
+  [[[(annotate/aggregation-name ag) (aggregation->rvalue ag)]]])
+
+(defn- group-and-post-aggregations
+  [id aggregations]
+  (let [expanded-ags (map expand-aggregation aggregations)
+        group-ags    (mapcat first expanded-ags)
+        post-ags     (mapcat second expanded-ags)]
+    [{$group (merge {"_id" id}
```

Comment:
I think this pattern is wrong and should be fixed everywhere we use it as it results in a vanilla map, rather than an ordered-map.

Reply:
yes, everything where order matters should be an ordered-map